lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -1031,10 +691,101 @@ static void createTMABarrierAndWait(
   }
 }
 
+// Similar to CoarseSchedule::insertDepsOfOp, we set <stage, cluster>
```

Comment:
Regarding al of the code below - what do you think if at the beginning of this pass we would have something like `coarseSchedule.deserialize(forOp)`, and we could then reuse all of the facilities we already have in CoarseSchedule, including dependency finding and schedule building?

Reply:
I think I had that in one of the versions of this diff. We can create a CoarseSchedule from the attributes then convert back to attributes. I can update the diff and we can then decide.